source/javascripts/pages/PipelinesPage/components/PipelineCanvas/GraphPipelineCanvas/components/GraphEdge.tsx

```diff
@@ -1,24 +1,43 @@
-import { BaseEdge, EdgeProps, getSmoothStepPath } from '@xyflow/react';
+import { CSSProperties } from 'react';
+import { BaseEdge, ConnectionLineComponentProps, EdgeProps, getSmoothStepPath, Position } from '@xyflow/react';
 
-type Props = EdgeProps;
+const edgeStyle = (style?: CSSProperties) => {
+  return { strokeWidth: 2, ...style };
+};
+
+const GraphEdge = (props: EdgeProps) => {
+  const { style } = props;
 
-const GraphEdge = (props: Props) => {
-  const { targetX, selected, style } = props;
-  const [edgePath] = getSmoothStepPath({ ...props, targetX, offset: 0, borderRadius: 12, centerX: targetX - 18 });
+  const [edgePath] = getSmoothStepPath({ ...props, offset: 0, borderRadius: 12 });
 
   return (
-    <BaseEdge
-      {...props}
-      path={edgePath}
-      interactionWidth={0}
-      style={{
-        ...style,
-        strokeWidth: 2,
-        pointerEvents: 'none',
-        stroke: selected ? 'var(--colors-border-strong)' : 'var(--colors-border-minimal)',
-      }}
-    />
+    <BaseEdge {...props} path={edgePath} style={edgeStyle({ ...style, stroke: 'var(--colors-border-minimal)' })} />
   );
 };
 
+const ConnectionGraphEdge = (props: ConnectionLineComponentProps) => {
+  const { fromX, fromY, toX, toY, connectionLineStyle, fromPosition, toPosition, connectionStatus } = props;
+
+  const targetXOffset = toPosition === Position.Left ? -8 : 8;
+  const sourceXOffset = fromPosition === Position.Left ? -8 : 8;
+
+  const stroke = ['valid', null].includes(connectionStatus)
```

Comment:
Do we want to add our own validation logic later? Now it's considered valid if it's the same node (or any other edge resulting in a cyclic graph), or if the edge is already existing.